gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java

```diff
@@ -351,11 +358,21 @@
         } else {
           this.existingTargetTable = Optional.absent();
         }
-
```

Comment:
This constructor is too complex, and it's hard to follow what's going on here. You can look if it can be refactored and split into smaller functions. If the refactoring is big and touches a lot of code, you can send it as a separate PR, before you main change.

You can also install this IntelliJ plugin that will tell you if the function is too complex, right in the IDE: https://plugins.jetbrains.com/plugin/12159-codemetrics